From 3d18857a29e082fb6b763fd2d41563c4feb6d966 Mon Sep 17 00:00:00 2001
From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Fri, 24 Aug 2012 11:25:56 +0200
Subject: [PATCH] journald: fix building with XZ support disabled

uncompress_blob() is not defined when XZ support is disabled. Without this
patch building fails with
[...]
.../systemd-189/src/journal/journal-verify.c:69: undefined reference to `uncompress_blob'
[...]
---
 src/journal/journal-verify.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/journal/journal-verify.c b/src/journal/journal-verify.c
index 2401293..5d134bd 100644
--- a/src/journal/journal-verify.c
+++ b/src/journal/journal-verify.c
@@ -63,6 +63,7 @@ static int journal_file_object_verify(JournalFile *f, Object *o) {
                 h1 = le64toh(o->data.hash);
 
                 if (o->object.flags & OBJECT_COMPRESSED) {
+#ifdef HAVE_XZ
                         void *b = NULL;
                         uint64_t alloc = 0, b_size;
 
@@ -73,6 +74,9 @@ static int journal_file_object_verify(JournalFile *f, Object *o) {
 
                         h2 = hash64(b, b_size);
                         free(b);
+#else
+                        return -EPROTONOSUPPORT;
+#endif
                 } else
                         h2 = hash64(o->data.payload, le64toh(o->object.size) - offsetof(Object, data.payload));
 
-- 
1.7.10.4

